shared/libraries/config_protocol/include/config_protocol/config_server_component.h

```diff
@@ -145,4 +160,38 @@ inline BaseObjectPtr ConfigServerComponent::update(const ComponentPtr& component
 
     return nullptr;
 }
+
+inline std::vector<std::string> ConfigServerComponent::splitPath(const std::string& path)
+{
+    std::vector<std::string> result;
+
+    size_t start = 0;
+    size_t pos = path.find('.', start);
+
+    while (pos != std::string::npos)
+    {
+        result.push_back(path.substr(start, pos - start));
+        start = pos + 1;
+        pos = path.find('.', start);
+    }
+
+    const auto rem = path.substr(start, pos);
+    if (!rem.empty())
+        result.push_back(rem);
+
+    return result;
+}
+
+inline PropertyObjectPtr ConfigServerComponent::getNestedPropertyObject(const ComponentPtr& component, const StringPtr& path)
+{
+    PropertyObjectPtr obj = component;
+    const auto pathStr = splitPath(static_cast<std::string>(path));
+    for (const auto& p : pathStr)
+    {
+        obj = obj.getPropertyValue(p);
+    }
+
+    return obj;
+}
```

Comment:
Does the PropertyObjectImpl dot access not work in this case? I'd think that just obj.getPropertyValue("obj1.obj2") should do exactly this.